src/mongocxx/test/spec/entity.hh

```diff
@@ -43,11 +42,28 @@ class map {
     map(map&&) = default;
     map& operator=(map&&) = default;
 
-    void insert(const key_type& key, mongocxx::client&& client);
+    ~map() = default;
 
-    bool contains(const key_type& key) const;
+    template <typename Entity>
+    bool insert(const key_type& key, Entity&& e) {
+        _map.emplace(key, std::forward<Entity>(e));
```

Comment:
It doesn't, but it does return a `pair<iterator, bool>`. The `bool` is true if an insertion happened, i.e., the element was not in the map, and we had to insert it. I changed this function to return that value instead. The function `create_entities` now errors if a key already exists. 

The test runner now clears the entity map before each test file. Previously, it would overwrite any entities with the same name.